Fix sqlglot>=23.0.0 issue

RobinL opened the pull request:
Closes #2072, see that issue for details 
Closes #2075 
Closes #2076

ADBond commented:
I think there will also be similar updates needed to [SQLTask](https://github.com/moj-analytical-services/splink/blob/bfaf8d142f7aad03f281b6324dd530e202eaec9a/splink/pipeline.py#L26) and [_exact_match_colname](https://github.com/moj-analytical-services/splink/blob/bfaf8d142f7aad03f281b6324dd530e202eaec9a/splink/comparison_level.py#L59)

RobinL commented:
Oops - that's me being overly optimistic, and also forgetting the tests are running against old SQL glot so of course they pass. Good spot.

I've updated the other methods, and updated the lockfile.

You can see in CI that tests passed for both the old and the new lockfile so i think we should be covered.

ThomasHepworth reviewed (approved):
Thanks Robin, this looks great.

Just to confirm, I've tested the code against the following SQLglot versions:
* 23.0.0
* 19.0.0
* 13.0.0

Everything is working as expected.

RobinL commented on splink/comparison_level.py:
```diff
@@ -37,12 +37,9 @@ def _is_exact_match(sql_syntax_tree):
-    if identifiers[0] == identifiers[1]:
+    identifiers = list(sql_syntax_tree.find_all(Identifier))
+    idents_truncated = [i.this[:-2] for i in identifiers]
+    if idents_truncated[0] == idents_truncated[1]:
```
Good idea!
